src/Microsoft.CorrelationVector/CorrelationVector.cs

```diff
@@ -1,397 +1,122 @@
-﻿// Copyright (c) Microsoft Corporation. All rights reserved.
-// Licensed under the MIT License.
-
-using System;
-using System.Threading;
-using System.Globalization;
-
-namespace Microsoft.CorrelationVector
-{
-    /// <summary>
-    /// This class represents a lightweight vector for identifying and measuring
-    /// causality.
-    /// </summary>
-    public sealed class CorrelationVector
-    {
-        internal const byte MaxVectorLength = 63;
-        internal const byte MaxVectorLengthV2 = 127;
-        internal const byte BaseLength = 16;
-        internal const byte BaseLengthV2 = 22;
-
-        internal readonly string BaseVector = null;
-
-        private int extension = 0;
-
-        private bool immutable = false;
-
-        private static Random rng = new Random();
-
-        /// <summary>
-        /// This is the header that should be used between services to pass the correlation
-        /// vector.
-        /// </summary>
-        public const string HeaderName = "MS-CV";
-
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Microsoft.CorrelationVector
+{
+    /// <summary>
+    /// This interface represents the abstract class from which Correlation Vectors are derived.
+    /// Not to be confused with the cV Base, which is the string that represents the base of the vector.
+    /// </summary>
+    public abstract class CorrelationVector : ICorrelationVector
+    {
+        protected CorrelationVector()
+        {
+
+        }
+        public abstract string Value { get; }
+        public abstract string Base { get; }
+        public abstract int Extension { get; }

```

Comment:
Do we need the base and extension to be abstract? We can have a public getter and protected setter for them.

Reply:
Since these are public, I don't think they should be changed within the class.

Reply:
Also, adding protected set introduces build errors for various reasons:

- Adding a set to Value involves changing BaseVector, which is readonly.
- Adding a set to Base does the same thing.
- The reason why the value Extension and the (lower case) extension member exist separately is because Increment() writes to this.extension using "ref this.extension" which does not let you pass a property or an indexer to it.